src/main/java/io/ebean/dbmigration/ddlgeneration/platform/BaseTableDdl.java

```diff
@@ -335,7 +337,7 @@ protected void writeUniqueOneToOneConstraints(DdlWrite write, CreateTable create
       }
       String[] columnNames = {col.getName()};
       write.apply()
-        .append(platformDdl.alterTableAddUniqueConstraint(tableName, uqName, columnNames, Boolean.TRUE.equals(col.isNotnull())))
+        .append(platformDdl.alterTableAddUniqueConstraint(tableName, uqName, columnNames, Boolean.TRUE.equals(col.isNotnull()) ? null : columnNames))
```

Comment:
unfortunately SQLServer, Oracle & DB2 handle null values different than Postgres, MySql & H2.
They allow only one null value. So I add the nullable columns here to exclude them from the unique index. (not yet implemented for Oracle & DB2)